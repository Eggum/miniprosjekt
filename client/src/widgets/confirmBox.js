import {Component} from "react-simplified";
import * as React from "react";

export class ConfirmBox extends Component<{
    modalId: React.Node,
    modalHeader: React.Node,
    modalBody: React.Node,
    onClick: () => mixed,
    children?: React.Node
}> {
    render() {
        return (
            <div
                className="modal fade"
                id={this.props.modalId}
                tabIndex="-1"
                role="dialog"
                aria-labelledby="exampleModalLabel"
                aria-hidden="true"
            >
                <div className="modal-dialog" role="document">
                    <div className="modal-content">
                        <div className="modal-header">
                            <h5 className="modal-title" id="exampleModalLabel">
                                {this.props.modalHeader}
                            </h5>
                            <button
                                type="button"
                                className="close"
                                data-dismiss="modal"
                                aria-label="Close"
                            >
                                <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        <div className="modal-body">{this.props.modalBody}</div>
                        <div className="modal-footer">
                            <button
                                type="button"
                                className="btn btn-secondary"
                                data-dismiss="modal"
                            >
                                No
                            </button>
                            <button
                                type="button"
                                className="btn btn-primary"
                                data-dismiss="modal"
                                onClick={this.props.onClick}
                            >
                                Yes
                            </button>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}