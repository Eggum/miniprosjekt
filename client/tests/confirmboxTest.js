// @flow

import * as React from 'react';
import { ConfirmBox } from '../src/widgets/confirmBox.js';
import { shallow, mount, ShallowWrapper } from 'enzyme';

/**
 * Confirm box tests.
 */

describe('Confirm box tests', () => {
    const test = jest.fn();

    const wrapper: ShallowWrapper = shallow(
        <ConfirmBox
            modalId={1}
            modalHeader={'test'}
            modalBody={'test'}
            onClick={test}
        />
    );

    it('initially', () => {
        expect(typeof wrapper).toEqual('object');
        expect(wrapper.find('button.btn-secondary')).toHaveLength(1);
        expect(wrapper.find('button.close')).toHaveLength(1);
        expect(wrapper.find('button.btn-primary')).toHaveLength(1);
    });

    it('after clicking yes button', () => {
        wrapper.find('button.btn-primary').simulate('click');

        let instance = ConfirmBox.instance();
        expect(typeof instance).toEqual('object');

        expect(test.mock.calls.length).toEqual(1);
    });

    it('renders correctly', () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
