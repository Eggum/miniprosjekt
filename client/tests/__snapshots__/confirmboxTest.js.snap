// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm box tests renders correctly 1`] = `
"<div className=\\"modal fade\\" id={1} tabIndex=\\"-1\\" role=\\"dialog\\" aria-labelledby=\\"exampleModalLabel\\" aria-hidden=\\"true\\">
  <div className=\\"modal-dialog\\" role=\\"document\\">
    <div className=\\"modal-content\\">
      <div className=\\"modal-header\\">
        <h5 className=\\"modal-title\\" id=\\"exampleModalLabel\\">
          test
        </h5>
        <button type=\\"button\\" className=\\"close\\" data-dismiss=\\"modal\\" aria-label=\\"Close\\">
          <span aria-hidden=\\"true\\">
            Ã—
          </span>
        </button>
      </div>
      <div className=\\"modal-body\\">
        test
      </div>
      <div className=\\"modal-footer\\">
        <button type=\\"button\\" className=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">
          No
        </button>
        <button type=\\"button\\" className=\\"btn btn-primary\\" data-dismiss=\\"modal\\" onClick={[Function: mockConstructor]}>
          Yes
        </button>
      </div>
    </div>
  </div>
</div>"
`;
